### ISSUE TITLE
IncompleteSignature error when calling Cloudfront from Github action

### ISSUE THREAD

**Issue by shukriadams** [aws/aws-sdk-js#3313]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js/issues) and didn't find any solution
  
**Describe the bug**

When I try to connect to AWS Cloudflare via a Github action, I get an IncompleteSignature error. My code looks like

```
const AWS = require('aws-sdk'),
              process = require('process')

    AWS.config.update({ 
        accessKeyId: process.env.KEY, 
        secretAccessKey: process.env.SECRET 
    })

    const cloudfront = new AWS.CloudFront(),
        params = {
            DistributionId: process.env.CLOUDFRONT_DISTRIBUTION_ID, 
            InvalidationBatch: { 
                CallerReference: new Date().getTime().toString(),
                Paths: {
                    Quantity: 1,
                    Items: [
                        '/*',
                    ]
              }
            }
        }

    cloudfront.createInvalidation(params, function (err, data) {
        if (err)
            console.log(err, err.stack)
        else
            console.log(data)
    })
    ```

and this is my error log

{ IncompleteSignature: '/20200617/us-east-1/cloudfront/aws4_request' not a valid key=value pair (missing equal-sign) in Authorization header: 'AWS4-HMAC-SHA256 Credential=*** /20200617/us-east-1/cloudfront/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=*********************************************************'.
    at Request.extractError (/tmp/deploy/node_modules/aws-sdk/lib/protocol/rest_xml.js:53:29)
    at Request.callListeners (/tmp/deploy/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/tmp/deploy/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/tmp/deploy/node_modules/aws-sdk/lib/request.js:683:14)
    at Request.transition (/tmp/deploy/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/tmp/deploy/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /tmp/deploy/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/tmp/deploy/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/tmp/deploy/node_modules/aws-sdk/lib/request.js:685:12)
    at Request.callListeners (/tmp/deploy/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
  message:
   '\'/20200617/us-east-1/cloudfront/aws4_request\' not a valid key=value pair (missing equal-sign) in Authorization header: \'AWS4-HMAC-SHA256 Credential=*** /20200617/us-east-1/cloudfront/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, 
```

I am running this inside a Docker container built from

```
FROM alpine:3.10
RUN apk update &&\
    apk upgrade &&\
    apk add git &&\
    apk add bash &&\
    apk add sshpass=1.06-r0 &&\
    apk add curl ca-certificates openssh &&\
    apk add nodejs=10.19.0-r0 &&\
    apk add npm=10.19.0-r0 &&\
    npm install yarn@1.17.3 -g &&\
    npm install webpack@4.39.1 -g &&\
    npm install webpack-cli@3.3.6 -g &&\
    npm install minify@5.1.1 -g &&\
    npm install jspm@0.16x -g &&\
    npm install uglify-es@3.3.9 -g &&\
    npm install concat-cli@4.0.0 -g &&\
    npm install typescript@3.5.3 -g &&\
    npm install tslint@5.18.0 -g &&\
    mkdir -p /usr/keys &&\
    mkdir -p /usr/build 

WORKDIR /usr/build

```

But, this is where it gets weird ...

I can run the code above in a Vagrant VM locally, and it works.

I can also use the same setup above (NodeJS in Docker container from a Github action), to successfully push a file to S3, f.ex

```
    const s3 = new AWS.S3()
    s3.upload({
        Bucket: S3_BUCKET,
        Key: 'test.txt',
        Body: 'foo bar'
    }, (err, result) => {    })
```

So basically, this error happens only when running in a  Github Action, and only when trying to run a CloudFront command.

**Is the issue in the browser/Node.js?**
Node.js

**If on Node.js, are you running this on AWS Lambda?**
No

**Details of the browser/Node.js version**
v10.19.0

**SDK version number**
2.694.0

**To Reproduce (observed behavior)**
See above.

**Expected behavior**
cloudfront.createInvalidation successfully creates an invalidation.

**Screenshots**
n/a

**Additional context**
Just wanted to make this perfectly clear - this error happens only in a docker container when called from a github action.

**Comment by ajredniwja** [comment]
Hey @shukriadams thanks for reaching out to us with your issue can you please check for any white space in your creds?

**Comment by shukriadams** [comment]
Thanks @ajredniwja, that managed to home in on the problem - there was a whitespace in the KEY value. Strangely, this whitespace did not impact S3 access, only Cloudfront. Fixed with a simple. trim() on this end.
